proposed/2023/Add-dotnet-search.md

```diff
@@ -0,0 +1,100 @@
+
+<!-- Explain the proposal in sufficient detail with implementation details, interaction models, and clarification of corner cases. -->
+The `package search [search terms] [options]` command will have the following options 
+
+| Option | Function |
+|---------|:----------|
+| `--Source` | A list of packages to search |
+| `--ExactMatch` | Return exact matches only as a search result |
+| `--Verbose` | Displays a detailed ist of information for each package |
+| `--Prerelease` | Allow prerelease packages to be shown. |
+| `--IncludeDelisted` | Allow unlisted packages to be shown |
+| `--NonInteractive` | Do not prompt for user input or confirmations.|
+| `--Help` | Show command help and usage information |
+| `--Verbosity` | Display the amount of details in the output: normal, quiet, detailed. |
+| `--ConfigFile` | The NuGet configuration file. If not specified, the hierarchy of configuration files from the current directory will be used. |
+|||
+#### **Algorithm**
+1. Read and parse arguments from user
+2. Use the nuget.exe Search API to look for the specific term in the provided source. The search filter will be composed of the `--Source`, `Search term`, `--Prerelease`, `--IncludeDelisted` values.
+3. Based on `--Verbosity` value, provide with a list of packages with various verbosity
+* Verbosity
+    - **Quiet** : Each line would look as follows : 
+    
+            <Package Name> <Latest Package Version>
+    - **Normal** :
+
+                <Package Name> | <Latest Package Version> | <Amount of Downloads>
+                <Description of the Package(short form)>
+    - **Detailed** :
+
+                <Package Name> | <Latest Package Version> | <Amount of Downloads>
+                Deprecated : YES/NO | Vulnerable : YES/NO
+                <Description of the Package(Readme file)>
+                License URL : <URL>
+
+#### **New Option `ExactMatch`**
+* This new option will allow for users to be able to search and have only exact matches as an output. 
+* For example if a user uses `dotnet package search NuGet.CommandLine --Verbosity Quiet`
+
+        NuGet.CommandLine | 6.7.0
+        NuGet.CommandLine.XPlat | 6.7.0
+        NuGet.Commands | 6.7.0
+        NuGet.exe | 3.4.3
+        NuGet.Bootstrapper | 2.6.0
+        CommandLineParser20 | 2.0.0
+        NuGet.VerifyMicrosoftPackage | 1.0.0
+        NuGet.for.MSBuild | 2.1.0 ...
+* Using ``dotnet package search NuGet.CommandLine --ExactMatch --Verbosity Quiet`` on the other side will have the following output
+
+         NuGet.CommandLine | 6.7.0
+## Drawbacks
+
+<!-- Why should we not do this? -->
+
+## Rationale and alternatives
+
+<!-- Why is this the best design compared to other designs? -->
+<!-- What other designs have been considered and why weren't they chosen? -->
+<!-- What is the impact of not doing this? -->
+In nuget.exe there is `nuget list` command which does the same thing. However, customers would like this functionality to be available in dotnet.
```

Comment:
Similar to my other comment, do we have any signals that a search command would not meet their need?

Reply:
I think you make a good point. However, according to https://github.com/NuGet/Home/issues/6060 it looks like, the users want the functionality in dotnet.

Reply:
I had a look through that issue. I don't see anything saying that list is specifically needed. In fact, I feel like that issue reinforces the idea that the request was made before any `search` command was available. I still think that `search` without any kind of list should be sufficient.

Reply:
Yeah, that sounds good to me. @nkolev92 , what do you think?

Reply:
I agree with @zivkan.

list is implemented based on the V2 protocol, and based on what was available back then. 
Listing all packages is a super heavy operation with the feeds as they are and we don't need it at this point.